tribits/doc/guides/TribitsSystemMacroFunctionDocTemplate.rst

```diff
@@ -15,6 +15,13 @@ understand the internals of TriBITS.
 @MACRO:    tribits_adjust_package_enables() +
 @FUNCTION: tribits_append_forward_dep_packages() +
 @MACRO:    tribits_assert_read_dependency_vars() +
+@FUCNTION: tribits_external_package_append_upstream_target_link_libraries_get_name_and_vis() +
+@FUNCTION: tribits_external_package_add_find_upstream_dependencies_str() +
```

Comment:
@KyleFromKitware, also note that the function:

* `tribits_external_package_append_upstream_target_link_libraries_get_name_and_vis()`

is being renamed to:

* `tribits_external_package_get_dep_name_and_vis()`

 as part of #494 shown [here](https://github.com/bartlettroscoe/TriBITS/blob/494-tpl-dependencies/tribits/core/package_arch/TribitsPackageDependencies.cmake#L163).

Reply:
> FWIW, I would use `UpperCamelCase_UpperCamelCase` rather than `UpperCamelCase_lowerCamelCase`. That's the convention used by [`ExternalProject`](https://cmake.org/cmake/help/latest/module/ExternalProject.html) and [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html).

I am just so used to coding conventions where proper nouns for package names, namespaces, and modules use `UpperCamelCase` but verbs for function and macro names use `lowerCamelCase()`.  In C++, the function would be named `TribitisExtPkg::addFindUpstreamDependenciesStr()`.  So a direct mapping to CMake would be to replace `:` with `_` with `TribitisExtPkg_addFindUpstreamDependenciesStr()`.  I guess to make this more clear/formal we could use a double underscore `__` to replace `:` so it would be `TribitisExtPkg__addFindUpstreamDependenciesStr()`.  It is all what you are used to I guess.